src/features/Students/StudentsPage/index.jsx

```diff
@@ -58,35 +124,103 @@ const StudentsPage = () => {
   };
 
   useEffect(() => {
+    fetchData(); // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentPage, filters]);
+
+  const handleAction = async () => {
+    await handleEnrollments(enrollmentData, selectedRow.ccx_id);
+    fetchData();
+    close();
+  };
+
+  const handleOpenModal = () => {
+    dispatch({ type: 'OPEN_MODAL' });
+  };
+
+  const handleCloseModal = () => {
+    dispatch({ type: 'CLOSE_MODAL' });
+  };
+
+  const handleApplyFilters = async () => {
     fetchData();
-    /* eslint-disable-next-line react-hooks/exhaustive-deps */
-  }, [state.currentPage]);
+    handleCloseModal();
+  };
+
+  const handleCleanFilters = () => {
+    setFilters(initialFilterFormValues);
+    setCurrentPage(1);
+    fetchData();
+  };
 
   const handlePagination = (targetPage) => {
+    setCurrentPage(targetPage);
     dispatch({ type: 'UPDATE_CURRENT_PAGE', payload: targetPage });
     fetchData();
   };
 
-  const COLUMNS = useMemo(() => getColumns(), []);
-
   return (
     <Container size="xl">
       <h1>Students</h1>
+      <Modal
+        title="Filters"
+        open={state.filters.isOpen}
+        body={
+          <Filters filters={filters} setFilters={setFilters} />
+        }
+        onClose={handleCloseModal}
+        buttons={
+          [<Button onClick={handleApplyFilters} variant="light">Apply Filters</Button>]
+        }
+      />
+      <ActionRow>
+        <IconButton
+          src={MenuIcon}
+          iconAs={Icon}
+          alt="Menu"
+          onClick={handleOpenModal}
+          variant="secondary"
+          size="inline"
+        />
+        <h4>Filters</h4>
+        <OverlayTrigger
+          placement="top"
+          overlay={<Tooltip variant="light">Clean filters</Tooltip>}
+        >
+          <IconButton icon={faTrash} alt="filter" onClick={handleCleanFilters} variant="secondary" />
+        </OverlayTrigger>
+      </ActionRow>
       <StudentsTable
-        data={state.data}
+        data={state.data || []}
```

Comment:
@anfbermudezme what's the reason to include this condition `|| []`? data is already declared as an empty array in initialState.